src/middleware/TelemetryHandler.ts

```diff
@@ -66,21 +67,25 @@ export class TelemetryHandler implements Middleware {
 	 */
 	public async execute(context: Context): Promise<void> {
 		try {
-			let clientRequestId: string = getRequestHeader(context.request, context.options, TelemetryHandler.CLIENT_REQUEST_ID_HEADER);
-			if (clientRequestId === null) {
-				clientRequestId = generateUUID();
-				setRequestHeader(context.request, context.options, TelemetryHandler.CLIENT_REQUEST_ID_HEADER, clientRequestId);
-			}
-			let sdkVersionValue: string = `${TelemetryHandler.PRODUCT_NAME}/${PACKAGE_VERSION}`;
-			let options: TelemetryHandlerOptions;
-			if (context.middlewareControl instanceof MiddlewareControl) {
-				options = context.middlewareControl.getMiddlewareOptions(TelemetryHandlerOptions) as TelemetryHandlerOptions;
-			}
-			if (typeof options !== "undefined") {
-				const featureUsage: string = options.getFeatureUsage();
-				sdkVersionValue += ` (${TelemetryHandler.FEATURE_USAGE_STRING}=${featureUsage})`;
+			if (typeof context.request === "string" && context.request.indexOf(GRAPH_BASE_URL) !== -1) {
```

Comment:
this is not going to work for national deployments

Reply:
us gov, chinese cloud, germany, etc...
https://docs.microsoft.com/en-us/graph/deployments

Reply:
Will checking if the url has the keyword 'graph' cover all the cases? 
If yes, could there be a case where the url contains 'graph' but the telemetry shouldn't be passed?

Reply:
I did something similar for unit testing for the mock test auth provider for large files upload to avoid attaching the JWT on non-graph requests. The only issue I see with checking just for graph is that teams has a service called teamsgraph.something.com (don't remember the full hostname) and some services have graph in the API path.
So I think it might be safer to check for "graph" only in the hostname as opposed to the full URL. But it might still cause us issues down the road if AGS enables more pass-through than today...

An alternative could be to put all the "known graph hostnames" in a array and compare if we have a match with anything in the array. A bit more compute intensive and it'll require us to update the array as new national deployments come online. But definitively safer.